netcore/System.Private.CoreLib/src/System/Array.Mono.cs

```diff
@@ -221,106 +221,118 @@ static bool CanAssignArrayElement (Type source, Type target)
-				throw new NotSupportedException ("Array type can not be an open generic type");
-			if (et.IsByRef)
-				throw new NotSupportedException (SR.NotSupported_Type);
-			
-			return CreateInstanceImpl (et, lengths, null);
+			Array array;
+			fixed (int* pLengths = &lengths [0])
+				array = InternalCreate (runtimeType._impl.Value, lengths.Length, pLengths, null);
+			GC.KeepAlive (runtimeType);
+			return array;
 		}
 
-		public static Array CreateInstance (Type elementType, int[] lengths, int [] lowerBounds)
+		public static unsafe Array CreateInstance (Type elementType, int[] lengths, int[] lowerBounds)
 		{
 			if (elementType == null)
-				throw new ArgumentNullException ("elementType");
+				ThrowHelper.ThrowArgumentNullException (ExceptionArgument.elementType);
 			if (lengths == null)
-				throw new ArgumentNullException ("lengths");
+				ThrowHelper.ThrowArgumentNullException (ExceptionArgument.lengths);
 			if (lowerBounds == null)
-				throw new ArgumentNullException ("lowerBounds");
-
-			if (!(elementType.UnderlyingSystemType is RuntimeType rt))
-				throw new ArgumentException ("Type must be a type provided by the runtime.", "elementType");
-			if (rt.Equals (typeof (void)))
-				throw new NotSupportedException ("Array type can not be void");
-			if (rt.ContainsGenericParameters)
-				throw new NotSupportedException ("Array type can not be an open generic type");
-			if (rt.IsByRef)
-				throw new NotSupportedException (SR.NotSupported_Type);
-
-			if (lengths.Length < 1)
-				throw new ArgumentException ("Arrays must contain >= 1 elements.");
-
-			if (lengths.Length != lowerBounds.Length)
-				throw new ArgumentException ("Arrays must be of same size.");
+				ThrowHelper.ThrowArgumentNullException (ExceptionArgument.lowerBounds);
+			if (lengths.Length != lowerBounds!.Length)
+				ThrowHelper.ThrowArgumentException (ExceptionResource.Arg_RanksAndBounds);
+			if (lengths.Length == 0)
+				ThrowHelper.ThrowArgumentException (ExceptionResource.Arg_NeedAtLeast1Rank);
 
-			for (int j = 0; j < lowerBounds.Length; j ++) {
-				if (lengths [j] < 0)
-					throw new ArgumentOutOfRangeException ($"lengths[{j}]", "Each value has to be >= 0.");
-				if ((long)lowerBounds [j] + (long)lengths [j] > (long)Int32.MaxValue)
-					throw new ArgumentOutOfRangeException (null, "Length + bound must not exceed Int32.MaxValue.");
-			}
+			for (int i = 0; i < lengths.Length; i++)
+				if (lengths [i] < 0)
+					ThrowHelper.ThrowArgumentOutOfRangeException (ExceptionArgument.lengths, i, ExceptionResource.ArgumentOutOfRange_NeedNonNegNum);
 
-			if (lengths.Length > 255)
-				throw new TypeLoadException ();
+			RuntimeType? runtimeType = elementType.UnderlyingSystemType as RuntimeType;
+			if (runtimeType == null)
+				ThrowHelper.ThrowArgumentException (ExceptionResource.Arg_MustBeType, ExceptionArgument.elementType);
 
-			return CreateInstanceImpl (elementType, lengths, lowerBounds);
+			Array array;
+			fixed (int* pLengths = &lengths [0])
```

Comment:
Any reason why `&lengths [0]` is used instead of `lengths`?

Reply:
@Therzok copy-paste from CoreCLR 🙂 there is no difference though